pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ComplexTypeTransformer.java

```diff
@@ -163,11 +172,20 @@ private static Map<String, String> parsePrefixesToRename(TableConfig tableConfig
 
   @Override
   public GenericRow transform(GenericRow record) {
-    flattenMap(record, new ArrayList<>(record.getFieldToValueMap().keySet()));
-    for (String collection : _fieldsToUnnest) {
-      unnestCollection(record, collection);
+    try {
```

Comment:
cc: @Jackie-Jiang I remember you mentioned that try-catch will impact the performance? shall do if-else check first?

Codewise I think this is simpler.
```
if (_continueOnError) {
    flattenMap(record, new ArrayList<>(record.getFieldToValueMap().keySet()));
    for (String collection : _fieldsToUnnest) {
      unnestCollection(record, collection);
    }
} else {
  try {
    flattenMap(record, new ArrayList<>(record.getFieldToValueMap().keySet()));
    for (String collection : _fieldsToUnnest) {
      unnestCollection(record, collection);
    }
  } catch (Exception e) {
       record.putValue(GenericRow.INCOMPLETE_RECORD_KEY, true);
       LOGGER.debug("Caught exception while transforming complex types for record: {}", record.toString(), e);
  }
}
```

Or this try-catch?

Reply:
I can do that. It just looked like code was being duplicated and hence didn't do it

Reply:
@Jackie-Jiang AFAIK, compiler doesn't optimize the code inside the try block, but other than that there's no performance overhead for just using a try block. Let me know if you have experienced otherwise.

Reply:
The current way should be fine. I don't remember seeing extra overhead on try-catch

Reply:
Thanks @Jackie-Jiang for confirming! @KKcorps can you also update the change in this PR(https://github.com/apache/pinot/pull/9376/files) to make the code consistent?